src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java

```diff
@@ -636,6 +636,16 @@ protected Response response(DataverseRequestHandler hdl, User user) {
         }
     }
 
+    /***
+     * The preferred way of handling a request from an open API.
+     *
+     * @param hdl handling code block.
+     * @return HTTP Response appropriate for the way {@code hdl} executed.
+     */
+    protected Response response(DataverseRequestHandler hdl) {
+        return response(hdl, null);
```

Comment:
This passes a null user to the [response(DataverseRequestHandler hdl, User user)] method (https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java#L629) which will create the DataverseRequest object with a null user, and the injected HttpServletRequest.

Until now I have always seen that the DataverseRequest class is instantiated with a user, even if it is GuestUser, in order to perform permission checks on the actions to be carried out on authenticated API endpoints.

Without the calling user, the DataverseRequest object would only contain the injected HttpServletRequest, which as far asI know its purpose there is to know the user IP address, which I'm not sure is useful information for unauthenticated operations like those in _Info_ (if I'm not mistaken, these address checks are only used in certain areas and are not common in Dataverse).

If there are no specific reasons that I am not aware of for persisting the HttpServletRequest data and therefore use the DataverseRequest class in unauthenticated scenarios, I would not create a DataverseRequest for unauthenticated operations, and would call the required logic directly like [here](https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/api/Info.java#L72).

I'm almost certain that null users are never passed when instantiating a DataverseRequest object. Maybe other people can confirm this. I also think that allowing null users can cause future accidental NPE issues.

Reply:
If you look at the method, which you mention is reintroduced, the one introduced is not the same as the previous one. Please take a look at the old one here: https://github.com/IQSS/dataverse/blob/v5.12/src/main/java/edu/harvard/iq/dataverse/api/AbstractApiBean.java#L705

You can see then old method always passed a non-null user to the DataverseRequest object, coming from the old findUserOrDie method.

Reply:
>as far as I know its purpose there is to know the user IP address, which I'm not sure is useful information for unauthenticated operations like those in Info

Probably we can get more input about this on the next stand-up meeting. Right now, it is not used on these public endpoints, but should we keep it so we can have an IP for log/additional functionality on the future?  

> If you look at the method, which you mention is reintroduced, the one introduced is not the same as the previous one. Please take a look at the old one here

The new method is not intended to be the same than the older one, I added it as a way to expose the response method without the need of the user. My initial thought process was that this was not done previously because there were dependencies to `response(DataverseRequestHandler hdl)` in the past while the other APIs were migrated to the new functionality defined on `response(DataverseRequestHandler hdl, User user)`.

Reply:
Alternatively, we could delete the new `response()` method and rewrite the three methods to not use it.

Reply:
@GPortas @jp-tosca and I discussed this and decided to remove the new `response()` method in 8e69c6d.